### PR TITLE
cpu support for benchmarks/oss.py

### PR DESCRIPTION
**Pull request by yuanyuan0057**
# Before submitting

- [X] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [X] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
Fix issues of benchmark/oss.py when trying to run it from cpu. Make two changes:
1. set device_ids to None when calling DDP as required by [PyTorch DDP](https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html?highlight=distributeddataparallel#torch.nn.parallel.DistributedDataParallel)
2. remove max_allocated_memory when cpu is in use. For cuda, if not args.cpu will be true and max_allocated_memory will still be displayed

## PR review
Talked with @blefaudeux on the issue.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: benchmarks/oss.py [hunk 1]
```diff
@@ -211,10 +214,7 @@ def closure():
 
     training_stop = time.monotonic()
     img_per_sec = n_items / (training_stop - training_start) * args.epochs
-    max_memory = torch.cuda.max_memory_allocated(rank) / 2 ** 20
```
Comment (min-xu-ai) [inline benchmarks/oss.py:214]: should we still have this log when it is not on CPU?

File: benchmarks/oss.py [hunk 2]
```diff
@@ -104,7 +104,10 @@ def train(
         optimizer = model.sharded_optimizer
 
     else:
-        model = DDP(model, device_ids=[rank], find_unused_parameters=False)  # type: ignore
+        if args.cpu:
+            model = DDP(model, device_ids=None, find_unused_parameters=False)  # type: ignore
```
Comment (min-xu-ai) [inline benchmarks/oss.py:108]: perhaps conditionally set the device_ids's value to a variable and avoid duplicating the entire model = DDP() line in the if and else cases?

File: benchmarks/oss.py (hunk 1 above)
Comment (yuanyuan0057) [reply benchmarks/oss.py:214]: We already have that on line 210.:     
if not args.cpu:
        torch.cuda.synchronize(rank)
        max_memory = torch.cuda.max_memory_allocated(rank) / 2 ** 20
        logging.info(f"[{dist.get_rank()}] : Peak memory {max_memory:.1f}MiB")

[the remark this answers is not in the document]
Comment (yuanyuan0057) [reply benchmarks/oss.py:214]: Same as previous reply. It was already set previously in line 210:
    if not args.cpu:
        torch.cuda.synchronize(rank)
        max_memory = torch.cuda.max_memory_allocated(rank) / 2 ** 20
        logging.info(f"[{dist.get_rank()}] : Peak memory {max_memory:.1f}MiB")

[the remark this answers is not in the document]
Comment (blefaudeux) [reply benchmarks/oss.py:214]: true that, the diff presentation here is wrong, I'm not sure why it shows these lines as being deleted ?

[the remark this answers is not in the document]
Comment (yuanyuan0057) [reply benchmarks/oss.py:214]: Yes, I deleted this line. This line is after the one in line 210. So on cuda, this line was evaluated twice.

File: benchmarks/oss.py (hunk 2 above)
Comment (yuanyuan0057) [reply benchmarks/oss.py]: Sure. This is done in the second commit

**Review by blefaudeux — APPROVED** [review]
Looks good to me, thanks for the PR Yuanyuan !

**Pull request merged**